src/Map/Map.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useRef, useEffect, useState } from 'react'
+import styled from 'styled-components'
+import mapboxgl from 'mapbox-gl'
+
+import colors from '../colors'
+
+interface StartPosition {
+  long: number
+  lat: number
+  zoom: number
+}
+
+interface Size {
+  height: string
+  width: string
+}
+
+interface ClusterOptions {
+  clusterMaxZoom: number
+  clusterAreaRadius: number
+  clusterColor: string
+  clusterRadius: number
+  countFont: string[]
+  countFontSize: number
+  countFontColor: string
+  pointColor: string
+  pointRadius: number
+}
+
+export interface Props {
+  token: string
+  size?: Size
+  startPosition?: StartPosition
+  style?: string
+  sourceJson?: string
+  cluster: boolean
+  clusterOptions?: ClusterOptions
+}
+
+const Wrapper = styled('div')<Size>`
+  height: ${({ height }) => height};
+  width: ${({ width }) => width};
+`
+
+const Map: React.FC<Props> = (props) => {
+  const mapContainer = useRef(null)
+  const map = useRef<mapboxgl.Map | null>(null)
+
+  mapboxgl.accessToken = props.token
+
+  const height = props.size?.height || '800px'
+  const width = props.size?.width || '800px'
+
+  const long = props.startPosition?.long || -3.5
+  const lat = props.startPosition?.lat || 55
+  const zoom = props.startPosition?.zoom || 5
+
+  const style = props.style || 'mapbox://styles/mapbox/light-v11'
+  const source = props.sourceJson || ''
+  const cluster = props.cluster || false
+
+  const clusterMaxZoom = props.clusterOptions?.clusterMaxZoom || 14
+  const clusterAreaRadius = props.clusterOptions?.clusterAreaRadius || 50
+  const clusterColor = props.clusterOptions?.clusterColor || colors.black
+  const clusterRadius = props.clusterOptions?.clusterRadius || 20
+  const countFont = props.clusterOptions?.countFont || ['Open Sans Regular']
+  const countFontSize = props.clusterOptions?.countFontSize || 14
+  const countFontColor = props.clusterOptions?.countFontColor || colors.white
+  const pointColor = props.clusterOptions?.pointColor || colors.black
+  const pointRadius = props.clusterOptions?.pointRadius || 6
+
```

Comment:
also this is something i would look into abstracting, but as a round 1 is good

Reply:
is this better https://github.com/digicatapult/ui-component-library/pull/26/commits/3fc33c4b0c6a96df4c2472ff33d12ecbb7bdb193?